### PR TITLE
Fix TCP keepalive

### PR DESCRIPTION
**Pull request by yadij**
Setting TCP keep-alive flags at accept(2) time resolves issues with
client sockets timing out while waiting for the ::Server handler to run.

Also resolves a bug with FTP DATA connections not having keep-alive set.
These connections would truncate objects if the data transfer connection
paused for too long and became timed out by the network routing system.

### CODE REVIEW

**Review by rousskov — CHANGES_REQUESTED** [review]
Nice fix, thank you! I do not insist on any of the changes in this review except for one (marked as such).

**Comment by rousskov** [comment]
I adjusted PR description, primarily to fix spelling and improve formatting. The API adjustments are natural/obvious/ and do not have to be explicitly disclosed/justified IMO so I removed the last paragraph. Please adjust further if needed.

File: src/comm.h [hunk 1]
```diff
@@ -23,7 +23,7 @@ bool comm_iocallbackpending(void); /* inline candidate */
 int commSetNonBlocking(int fd);
 int commUnsetNonBlocking(int fd);
 void commSetCloseOnExec(int fd);
-void commSetTcpKeepalive(int fd, int idle, int interval, int timeout);
+void commSetTcpKeepalive(int fd, const AnyP::PortCfg::TcpKeepalive &);
```
[the remark this answers is not in the document]
Comment (yadij) [reply src/comm.h:26]: Done. Will push shortly.

File: src/comm.cc [hunk 2]
```diff
@@ -1143,29 +1143,32 @@ commSetTcpNoDelay(int fd)
 #endif
 
 void
-commSetTcpKeepalive(int fd, int idle, int interval, int timeout)
+commSetTcpKeepalive(int fd, const AnyP::PortCfg::TcpKeepalive &cfg)
 {
+    if (!cfg.enabled)
+        return;
```
[the remark this answers is not in the document]
Comment (yadij) [reply src/comm.cc:1149]: Done. will push shortly

**Comment by rousskov** [comment]
I pushed a few polishing changes. Please adjust as needed. If I get a chance before this is merged, I will self-request a review and polish the moved code further.

**Comment by yadij** [comment]
Additions so far look good to me.

**Review by rousskov — COMMENTED** [review]
I am done modifying the code. Please adjust further as needed, of course. My commit 16badfa implies that this code was not properly tested. Please do test your PRs!

**Pull request closed without merging**